cognite_toolkit/cdf_tk/templates/_templates.py

```diff
@@ -377,74 +416,107 @@ def sort_key(p: Path) -> int:
         # The custom key 'sort_key' is to get the sort on integer and not the string.
         filepaths = sorted(filepaths, key=sort_key)
 
+        # Initialise for auth, other resource folders will be added as they are found
+        all_files: dict[str, dict[str, list[Path]]] = {
+            "auth": {
+                "resource_files": [],
+                "other_files": [],
+            }
+        }
```

Comment:
This makes the code hard to read for me, I think there are two improvement that can be done.

1. I would rename `all_files` to `files_by_resource_folder` making it clear what this is.
2. I would introduce a helper data class instead of the dictionary

```python
@dataclass
class ResourceFiles:
    resource: list[Path] = field(default_factory=dict)
    other: list[Path] = field(default_factory=dict)
```

Then you can write this as:
```suggestion
        files_by_resource_folder : dict[str, ResourceFiles] = defaultdict(ResourceFiles)
        files_by_resource_folder ["auth"] = ResourceFiles()
```

**Remark** Why to you want to ensure that `auth` is initialized? Is it necessary?

Reply:
mypy complains...